pkg/controller/machine_util.go

```diff
@@ -496,3 +504,51 @@ func SyncMachineTaints(
 
 	return toBeUpdated
 }
+
+func (c *controller) UpdateNodeTerminationCondition(machine *v1alpha1.Machine) error {
+	if machine.Status.CurrentStatus.Phase == "" {
+		return nil
+	}
+
+	nodeName := machine.Status.Node
+
+	terminationCondition := v1.NodeCondition{
```

Comment:
Just something to mention, now we have a node-condition which only pops up during the machine-deletion. 

I believe, node-conditions are generally maintained throughout the life of the machine/resource, and updated with the current state.
Would be a little weird to have it while the machine is running as well, as it's not really terminated. I cant see, anything that needs to be done with this PR.
cc @amshuman-kr

Reply:
@hardikdr Good point. I am not sure about the behaviour of custom node conditions maintained by NPD, if they are maintained throughout node life-cycle or maintained when the condition occurs. Might be good check.

Regarding maintaining the termination condition throughout the node life-cycle, if we call the type `Termination` then it doesn't sound right to maintain it throughout. But if we name if something like `SelectedForTermination`, then it might still be ok to maintain it throughout and not just during machine deletion.

I am not proposing that we already change this PR to maintain the condition throughout the node life-cycle. Merely to rename the condition type to work both when we add it momentarily during machine deletion and when we want to change in the future to maintain it always.

WDYT?